### DISCUSSION TITLE
InstanceID changes after local server restart

### DISCUSSION THREAD

**Discussion by barashF** [SmartlyDressedGames/Unturned-3.x-Community#4052] [Q&A] [answered]
To determine a specific instance, I use instanceID, but sometimes after restarting the server for machines, it can change barricades. Please tell me, is this due to an incorrect shutdown of the local server, or is it supposed to change?

**Comment by DanielWillett** [comment, marked as answer]
It doesn't change... are you by any chance using the unity method, GetInstanceId? (you shouldn't be)

BarricadeDrop.instanceId is the right property.

**Reply by barashF** [reply]
And for vehicles should interactablevehicle.instanceID? But why getinstanceID shouldn't be?

**Reply by GazziFX** [reply]
They are absolutely different, GetInstanceId is inherited from UnityEngine.Object and used to reference components in some structs like RaycastHit
